flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,329 @@
+
+Assuming an honest and aware payer, the payer would realize their transaction failed when their balance first dropped below minimum account balance. They would then top off their account before continuing to send more transactions.
+
+With an unaware payer (for example due to bad automation) or a malicious payer, that wants to drain the access node account or wants to cause excess traffic without paying transaction fees, it is in the access nodes best interest to not include these transactions.
+
+The access node can check the balance of the payer prior to including their transaction and if the balance is below the minimum account balance, it would not include the transaction. However this prevents the payer from topping up his own balance, after an hones mistake. Some thought is needed here to address this edge case.
+
+The problem of providing economic incentive to include transactions of not part of this FLIP. This FLIP only provides a way to discourage access nodes to include transactions that cannot be paid for by the payer. 
+
+### Failing transactions
+
+There are a few reasons why a transaction fails, which can be split into four categories, according to who finally gets charged for the transaction fees and how much they get charged.
+
+1. Transaction failed because the payers signature was incorrect, or the payer does not have enough funds to cover maximum transaction costs (transaction costs with the execution effort set to the execution effort limit).
+
+    This transaction should have been rejected at the access node level. However since it was not and there is no way to charge the payer, the access node account is charged with the transaction fees. The transaction fees are computed with the execution effort set to 0.
+    
+2. Transaction failed before the transaction script execution started (some non-payer signature was incorrect, or the sequence number was incorrect)
+
+    In this scenario the payer is charged for the transaction fees with the execution effort set to 0.
+
+3. Transaction failed during transaction script parsing or during script execution, during fee deduction or during the storage used check. 
+
+    In this case, the fees are charged normally with the execution effort set to the actual usage before the error was produced.
+
+4. Transaction failed because the execution effort limit was reached.
+
+    For these transactions the payer is charged with the transaction fees where the execution effort is set to the execution effort limit.
+
+### Parameter calibration
+
+When turning on variable transaction fees the goal is to have a smooth transition to the new system. To help facilitate this, the cost of transaction fees should be set so that ~95% of transactions should actually pay less transaction fees with variable transaction fees turned on. The two different parts of variable transaction fees need to be calibrated accordingly.
```

Comment:
```suggestion
When turning on variable transaction fees, a smooth transition to the new system is desired. To help facilitate this, the cost of transaction fees should be set so that ~95% of transactions should actually pay less transaction fees with variable transaction fees turned on. The two different parts of variable transaction fees need to be calibrated accordingly.
```